From 875c2c5d91c1d316e3dedd386b62fa003b7cec9e Mon Sep 17 00:00:00 2001
From: Andreas Holzammer <andreas.holzammer@kdab.com>
Date: Fri, 14 Jan 2011 15:48:37 +0100
Subject: [PATCH] fix widget creation
 If a window which should not be shown has
 a height and width, its gets shown as a black
 window under wince

---
 src/gui/kernel/qwidget_wince.cpp |    3 ++-
 1 files changed, 2 insertions(+), 1 deletions(-)

diff --git a/src/gui/kernel/qwidget_wince.cpp b/src/gui/kernel/qwidget_wince.cpp
index 76532ed..fb3d31f 100644
--- a/src/gui/kernel/qwidget_wince.cpp
+++ b/src/gui/kernel/qwidget_wince.cpp
@@ -283,7 +283,8 @@ void QWidgetPrivate::create_sys(WId window, bool initializeWindow, bool destroyO
 
         // Adjust for framestrut when needed
         RECT rect = {0,0,0,0};
-        if (AdjustWindowRectEx(&rect, style, FALSE, exsty)) {
+        bool isVisibleOnScreen = !q->testAttribute(Qt::WA_DontShowOnScreen);
+        if (isVisibleOnScreen && AdjustWindowRectEx(&rect, style, FALSE, exsty)) {
             QTLWExtra *td = maybeTopData();
             if (wasMoved && (td && !td->posFromMove)) {
                 x = data.crect.x() + rect.left;
-- 
1.7.0.2.msysgit.0

